odp_sdk/tabular_storage_client.py

```diff
@@ -254,15 +267,128 @@ def delete_stage_request(self, resource_dto: ResourceDto, table_stage: TableStag
         except requests.HTTPError as e:
             if response.status_code == 400:  # This is coming as 400 for both 400 and 404 from ODP
                 raise OdpResourceNotFoundError("Schema not found") from e
+            raise
+
+    def select(self, resource_dto: ResourceDto, filter_query: Optional[dict]) -> List[Dict]:
```

Comment:
Method does not handle pagination.

Suggest changing the signature to:
```python
def select(self, resource_dto: ResourceDto, filter_query: Optional[dict], limit: Optional[int] = None, stream: bool = False) -> Iterable[dict]:
    ...
```

If `stream=True`, the rows are yielded row-by-row as a `iterable[dict]`, while if `stream=False`, the rows are assembled and return as a `list[dict]`.

The function should handle pagination, so it could be helpful to have a second function to make the actual API-call with the following signature:

```python
def select_page(self, resource_dto: ResourceDto, filter_query: Optional[dict], limit: Optional[int] = None, cursor: Optional[str] = None) -> tuple[list[dict], Optional[str]]:
    ...
```

This function will make a single API request for a given page and return the rows together with

Then having another function to handle streaming:

```python
def select_stream(self, resource_dto: ResourceDto, filter_query: Optional[dict], limit: Optional[int] = None, cursor: Optional[str] = None) -> Iterator[dict]:
    
    page_size = limit  # Maybe have smaller page size?

    while True:
        rows, cursor = select_page(resource_dto, filter_query, limit, cursor)
        yield from rows

        # Handle limit if page size is less than limit-argument

        if cursor is None:
            # Break once we reached the end of the table
            break
```

The `select`-function then generalizes quite nicely:

```python
def select(self, resource_dto: ResourceDto, filter_query: Optional[dict], limit: Optional[int] = None, stream: bool = False) -> Iterable[dict]:
    row_iterator = select_stream(resource_dto, filter_query, limit)
    
    if not stream:
        return list(row_iterator)
    
    yield from row_iterator
```